metaseq/data/atomic_array.cpp

```diff
@@ -0,0 +1,235 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <iostream>
+
+static PyObject* kwnames;
+static PyObject* format;
+struct Owned {
+    Owned(PyObject* o)
+    : obj(o) {}
```

Comment:
Will take this opportunity to learn about this syntax 😅 : what does `obj(o)` do here as part of the constructor member initialization?  And why is the destructor only decrementing the reference count for `obj`?

Reply:
Essentially it does `obj = o`. C++ has to provide an initial value to `obj` before the body of the function starts so that objects in the class otherwise all objects would need zero-argument default constructors.